app/models/person.rb

```diff
@@ -121,6 +146,76 @@ def self.parse_message(message: nil)
       description: description,
       links: links,
       identifiers: identifiers,
-      country: country }.compact)
+      country: country,
+      employment: employment }.compact)
+  end
+
+  # SPARQL query to fetch organizational identifiers from Wikidata
+  # PREFIX wikibase: <http://wikiba.se/ontology#>
+  # PREFIX wd: <http://www.wikidata.org/entity/> 
+  # PREFIX wdt: <http://www.wikidata.org/prop/direct/>
+  # PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+  # PREFIX p: <http://www.wikidata.org/prop/>
+  # PREFIX v: <http://www.wikidata.org/prop/statement/>
+
+  # SELECT ?item ?itemLabel ?ror ?grid ?ringgold WHERE {  
+  #   ?item wdt:P6782 ?ror;
+  #         wdt:P3500 ?ringgold;
+  #         wdt:P2427 ?grid .
+  #   FILTER(?ringgold in ("9177", "219874", "14903") || ?grid in ("grid.475826.a")).
+  #   SERVICE wikibase:label {
+  #     bd:serviceParam wikibase:language "[AUTO_LANGUAGE]" .
+  #   }
+  # }
+  def self.wikidata_query(employment)
+    ringgold_filter = Array.wrap(employment).reduce([]) do |sum, f|
+      sum << f["ringgold"] if f["ringgold"]
+
+      sum
+    end.join("%22%2C%20%22")
+    ringgold_filter = "%3Fringgold%20in%20(%22#{ringgold_filter}%22)" if ringgold_filter.present?
+
+    grid_filter = Array.wrap(employment).reduce([]) do |sum, f|
+      sum << f["grid"] if f["grid"]
+
+      sum
+    end.join("%22%2C%20%22")
+    grid_filter = "(%3Fgrid%20in%20(%22#{grid_filter}%22)" if grid_filter.present?
+    filter = [ringgold_filter, grid_filter].compact.join("%20%7C%7C%20")
+
+    url = "https://query.wikidata.org/sparql?query=PREFIX%20wikibase%3A%20%3Chttp%3A%2F%2Fwikiba.se%2Fontology%23%3E%0APREFIX%20wd%3A%20%3Chttp%3A%2F%2Fwww.wikidata.org%2Fentity%2F%3E%20%0APREFIX%20wdt%3A%20%3Chttp%3A%2F%2Fwww.wikidata.org%2Fprop%2Fdirect%2F%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20p%3A%20%3Chttp%3A%2F%2Fwww.wikidata.org%2Fprop%2F%3E%0APREFIX%20v%3A%20%3Chttp%3A%2F%2Fwww.wikidata.org%2Fprop%2Fstatement%2F%3E%0A%0ASELECT%20%3Fitem%20%3FitemLabel%20%3Fror%20%3Fgrid%20%3Fringgold%20WHERE%20%7B%20%20%0A%20%20%3Fitem%20wdt%3AP6782%20%3Fror%3B%0A%20%20%20%20%20%20%20%20wdt%3AP3500%20%3Fringgold%3B%0A%20%20%20%20%20%20%20%20wdt%3AP2427%20%3Fgrid%20.%0A%20%20FILTER(#{filter})).%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%0A%20%20%20%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%22%20.%0A%20%20%20%7D%0A%7D"
```

Comment:
Is it possible to build the query and encode the url with ruby, I assume it matches what it is in the comment, but it is quite difficult to make changes to this query pre escaped.

Alternatively did you consider using a sparql client like: https://github.com/ruby-rdf/sparql-client

Reply:
I am planning to do that, but first wanted to get the basic functionality working. But I can refactor in this pull request. I will try the Ruby SPARQL client.